p2p/protocol.go

```diff
@@ -0,0 +1,202 @@
+package p2p
+
+import (
+	"container/list"
+	"errors"
+	"github.com/gogo/protobuf/proto"
+	"github.com/spacemeshos/go-spacemesh/crypto"
+	"github.com/spacemeshos/go-spacemesh/log"
+	"github.com/spacemeshos/go-spacemesh/p2p/pb"
+	"github.com/spacemeshos/go-spacemesh/p2p/service"
+	"sync"
+	"sync/atomic"
+	"time"
+)
+
+type MessageType uint32
+
+type Item struct {
+	id        uint64
+	timestamp time.Time
+}
+
+type Protocol struct {
+	id                 uint64
+	name               string
+	network            Service
+	pendMutex          sync.RWMutex
+	pendingMap         map[uint64]chan interface{}
+	pendingQueue       *list.List
+	resHandlers        map[uint64]func(msg []byte)
+	msgRequestHandlers map[MessageType]func(msg []byte) []byte
+	ingressChannel     chan service.Message
+	requestLifetime    time.Duration
+}
+
+func NewProtocol(network Service, name string, ttl time.Duration) *Protocol {
```

Comment:
document the method, not sure if the purpose of 'ttl' is clear to the user of this class

Reply:
added documentation to the struct and change the var name in the constractor